src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandle.T.cs

```diff
@@ -0,0 +1,132 @@
+        /// Returns the internal integer representation of a <see cref="GCHandle{T}"/> object.
+        /// </summary>
+        /// <param name="value">A <see cref="GCHandle{T}"/> object to retrieve an internal integer representation from.</param>
+        /// <returns>An <see cref="IntPtr"/> object that represents a <see cref="GCHandle{T}"/> object.</returns>
+        /// <remarks>
+        /// The <see cref="IntPtr"/> representation of <see cref="GCHandle{T}"/> is not
+        /// interchangable with <see cref="GCHandle"/>.
+        /// </remarks>
+        public static IntPtr ToIntPtr(GCHandle<T> value) => value._handle;
+
+        /// <summary>
+        /// Releases this <see cref="GCHandle{T}"/>.
+        /// </summary>
+        public void Dispose()
+        {
+            // Free the handle if it hasn't already been freed.
+            // Unlike GCHandle.Free, no thread safety is provided.
+            IntPtr handle = _handle;
+            if (handle != IntPtr.Zero)
+            {
+                _handle = IntPtr.Zero;
+                GCHandle.InternalFree(handle);
+            }
+        }
+
+        /// <summary>Indicates whether the current instance is equal to another instance of the same type.</summary>
+        /// <param name="obj">An instance to compare with this instance.</param>
+        /// <returns>true if the current instance is equal to the other instance; otherwise, false.</returns>
+        public override readonly bool Equals([NotNullWhen(true)] object? obj) => obj is GCHandle<T> handle && Equals(handle);
+
+        /// <summary>Indicates whether the current instance is equal to another instance of the same type.</summary>
+        /// <param name="other">An instance to compare with this instance.</param>
+        /// <returns>true if the current instance is equal to the other instance; otherwise, false.</returns>
+        public readonly bool Equals(GCHandle<T> other) => _handle == other._handle;
+
+        /// <summary>
+        /// Returns an identifier for the current <see cref="GCHandle{T}"/> object.
+        /// </summary>
+        /// <returns>An identifier for the current <see cref="GCHandle{T}"/> object.</returns>
+        public override readonly int GetHashCode() => _handle.GetHashCode();
+
+        /// <summary>
+        /// Returns a value indicating whether two <see cref="GCHandle{T}"/> objects are equal.
+        /// </summary>
+        /// <param name="left">A <see cref="GCHandle{T}"/> object to compare with the <paramref name="right"/> parameter.</param>
+        /// <param name="right">A <see cref="GCHandle{T}"/> object to compare with the <paramref name="left"/> parameter.</param>
+        /// <returns>
+        /// <see langword=""/> if the <paramref name="left"/> and <paramref name="right"/> parameters are equal; otherwise, <see langword="false"/>.
+        /// </returns>
+        public static bool operator ==(GCHandle<T> left, GCHandle<T> right) => left._handle == right._handle;
```

Comment:
The operators are not part of the approved API surface.

Reply:
I added them as a group though 1 click in VS. There are structs implementing `IEquatable<T>` only so it will be fine.

Reply:
Adding an unapproved API surface is not fine.

Reply:
I meant it's fine to not have equality operators.
Not overriding `Equals(object)` and `GetHashCode` will cause CA1067, and the should have less public API impact because C# compiler will always use `callvirt` for them. Thus I assume the overrides should be preserved.

Reply:
> Equals(object) and GetHashCode

Yes, these are fine - they are part of `IEquatable<self>` contract.